tests/legacy-cli/e2e/tests/misc/module-resolution.ts

```diff
@@ -38,7 +46,7 @@ export default async function () {
   await ng('build', '--aot');
   await ng('test', '--watch=false');
 
-  await silentNpm('install', 'firebase@4.9.0');
+  await silentNpm('install', 'firebase@4.13.1');
```

Comment:
In `4.9.0` firebase used to publish typescript files in their dist. Which causes the below error;
```
Module build failed: Error: /tmp/angular-cli-e2e-118519-141-12iaj2q.ncmw/test-project/node_modules/@firebase/polyfill/index.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
The missing file seems to be part of a third party library. TS files in published libraries are often a sign of a badly packaged library. Please open an issue in the library repository to alert its author and ask them to package the library using the Angular Package Format (https://goo.gl/jB3GVv).
    at AngularCompilerPlugin.getCompiledFile (/tmp/angular-cli-e2e-118519-141-12iaj2q.ncmw/test-project/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:730:23)
    at plugin.done.then (/tmp/angular-cli-e2e-118519-141-12iaj2q.ncmw/test-project/node_modules/@ngtools/webpack/src/loader.js:41:31)
```

https://unpkg.com/@firebase/polyfill@0.1.6/index.ts

This was fixed with this PR: https://github.com/firebase/firebase-js-sdk/pull/695

Reply:
That version is intentionally chosen for the reason described.  The bundler's package resolution to determine the actual javascript file to use within the application should still function in this case.

Reply:
So, ngtools/webpack should have a different resolution than ngc to cater for such instances?

Reply:
It's also important that the appropriate bundle format is chosen from the `package.json` (FESM2015, ESM2015, etc.) which is something that TypeScript has no knowledge when resolving.

Reply:
The resolution itself of the module formats still happends via webpack, as to webpack I am passing the directory that contains the package.json and only fallback to the js file if it doesn’t exists. (Same as current basicilly)